sktime/datatypes/_dtypekind.py

```diff
@@ -0,0 +1,116 @@
+from enum import IntEnum
+
+import numpy as np
+import pandas as pd
+from pandas.api.types import (
+    is_bool_dtype,
+    is_datetime64_any_dtype,
+    is_float_dtype,
+    is_signed_integer_dtype,
+    is_string_dtype,
+    is_unsigned_integer_dtype,
+)
+
+
+class DtypeKind(IntEnum):
+    """
+    Integer enum for data types.
+
+    Attributes
+    ----------
+    INT : int
+        Matches to signed integer data type.
+    UINT : int
+        Matches to unsigned integer data type.
+    FLOAT : int
+        Matches to floating point data type.
+    BOOL : int
+        Matches to boolean data type.
+    STRING : int
+        Matches to string data type (UTF-8 encoded).
+    DATETIME : int
+        Matches to datetime data type.
+    CATEGORICAL : int
+        Matches to categorical data type.
+    """
+
+    INT = 0
+    UINT = 1
+    FLOAT = 2
+    BOOL = 20
+    STRING = 21  # UTF-8
+    DATETIME = 22
+    CATEGORICAL = 23
+
+
+def _pandas_dtype_to_kind(col_dtypes):
+    for i, dtype in enumerate(col_dtypes):
```

Comment:
I would simplify and add a docstring.

* simplify: primarily assume a simple index field, not a list
* at the start, you can treat list by recursion. If `isinstance(col_dtypes, list): return [_pandas_dtype_to_kind(x) for x in col_types]`
* add a docstring while you still know what this does. This is good to give to AI (copilot etc) and then minimally iterate